Assets/Scripts/Models/HeadlineGenerator.cs

```diff
@@ -0,0 +1,81 @@
+#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Xml;
+
+public class HeadlineGenerator 
+{
+    private List<string> headlines = new List<string>();
+    private float time, nextTime, minInterval, maxInterval;
+
+    public HeadlineGenerator(XmlNode baseNode)
+    {
+        // TODO Consider default values for these. Also consider reasonable limits
+        minInterval = float.Parse(baseNode.Attributes.GetNamedItem("minInterval").Value);
```

Comment:
If you want random delay times with Scheduler you can do it by making the event one shot and having the event callback queue up the next event. Maybe one day Scheduler will have a nicer API to automate this, but it is possible now.